Use version number in notebook cache directories

dster2 opened the pull request:
Despite the fact that neither Datasets nor Notebooks datasources currently support multi-versioning when attached to a Kaggle Notebook Session, our existing Dataset support in HTTP resolving *does* use a cache dir with the version number so that multiple versions are supported in that mode.  Here we update the recent Notebook support to match that behavior.

Some minor tangentially related cleanup for consistency reasons too.

http://b/384702998

neshdev reviewed (approved):
LGTM. Anyway you can provide testing evidence? It seems tat the integration should cover it. Normally, we at least have 1 notebook on  kaggle to ensure everything is working as expected.

dster2 commented:
@neshdev Thanks for the review, could you please clarify what you'd like added?  I added an integration test which should cover the support added here, as I think you noted, but then I'm not sure what you're looking for?

Indeed I forgot the bug, added that.

neshdev commented:
You can take a look at this [PR](https://github.com/Kaggle/kagglehub/pull/141#issuecomment-2214885969). Basically, just a kaggle notebook with kagglehub installed from from current branch.

This is an example: [b/351825012](https://b.corp.google.com/issues/351825012), but maybe overkill.

dster2 commented:
Ah very cool, thanks.  Sorry for archiving that bug, I made some mistakes on Taskflow 😅

But unfortunately the functionality added here does NOT work within Kaggle, the Kaggle Cache does not support multiple simultaneous versions of the same Notebook (or Dataset) currently (only for Models) though we would like to add that in the future.  Nonetheless we do support multi-versioning with the HTTP Resolver for Datasets, and I thought Notebooks should match that behavior (and it's a nice touch for Packages).

neshdev commented:
I see. We can review more during our knowledge sharing session. LGTM if its working as expected for you.
